src/main/java/xyz/holocons/mc/holoitemsrevamp/enchant/EnchantListener.java

```diff
@@ -57,6 +59,22 @@ public void onBlockBreak(BlockBreakEvent event) {
         });
     }
 
+    /**
+     * Handles BlockPlace enchantments.
+     *
+     * @param event The BlockBreakEvent
+     */
+    @EventHandler(ignoreCancelled = true)
+    public void onBlockPlace(BlockPlaceEvent event) {
+        final var itemStack = event.getPlayer().getInventory().getItemInMainHand();
```

Comment:
This is incorrect. Blocks can be placed from either the main hand or offhand. Use `BlockPlaceEvent::getItemInHand` to get it.